### PR TITLE
[PS-2558] Mobile Automation - Starting automationIDs additions to our codebase

### PR DESCRIPTION
**Pull request by ifernandezdiaz**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X] Other

## Objective
To improve the reliability of our automated mobile tests, we added AutomationIDs for the most important elements in our mobile app.

## Code changes
* **EnvironmentPage.xaml:** Locators added for URL fields
* **HomePage.xaml:** Locators added for `Email` field, `Continue` and `Create Account` buttons  
* **LockPage.xaml:** Locators added for `Unlock` and `Toggle Visibility` buttons, apart from `PIN` and `Password` fields
* **LoginPage.xaml:** Locators added for each interactable element.
* **RegisterPage.xaml:**  Locators added for each interactable element.

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/App/Pages/Accounts/RegisterPage.xaml [hunk 1]
```diff
@@ -59,7 +60,8 @@
                         IsTextPredictionEnabled="False"
                         IsPassword="{Binding ShowPassword, Converter={StaticResource inverseBool}}"
                         Grid.Row="1"
-                        Grid.Column="0" />
+                        Grid.Column="0"
+                        AutomationId="MasterPasswordField"/>
```
Comment (vvolkgang) [inline src/App/Pages/Accounts/RegisterPage.xaml:64]: 🤔 some of these have a suffix related to the item type, most don't. Can we define a standard so our devs can follow it too? It should:

1. Define a name for each element type
    * Button -> Button
    * Switch, IconButton -> Toggle
    * Label -> Label
    * Entry, MonoEntry -> Input
    * ...
1. Define if we prefix or suffix those names. `ButtonLoginWithDevice` vs `LoginWithDeviceButton`
1. Be relevant for you and your team, you don't need to follow the same code element names. They should still make sense even if the implementation changes.
    
That list is just an example and I'm not trying to be prescriptive about nomenclature, feel free to define what works better for you / your team!

**Review by vvolkgang — CHANGES_REQUESTED** [review]

File: src/App/Pages/Accounts/RegisterPage.xaml (hunk 1 above)
Comment (ifernandezdiaz) [reply src/App/Pages/Accounts/RegisterPage.xaml:64]: Hey! Yes, this is my fault! I planned to use the same standard on all the IDs I added, but I didn't. It'll be fixed on my next commit 💪 

Also, I'll explain the standard on the document I mentioned yesterday 🙌

File: src/App/Pages/Accounts/EnvironmentPage.xaml [hunk 2]
```diff
@@ -53,7 +54,8 @@
                         x:Name="_webVaultEntry"
                         Text="{Binding WebVaultUrl}"
                         Keyboard="Url"
-                        StyleClass="box-value" />
+                        StyleClass="box-value"
+                        AutomationId="WebVaultURL"/>
```
[the remark this answers is not in the document]
Comment (ifernandezdiaz) [reply src/App/Pages/Accounts/EnvironmentPage.xaml]: Done!

**Review by vvolkgang — APPROVED** [review]
LGTM!

File: src/App/Pages/Accounts/EnvironmentPage.xaml [hunk 3]
```diff
@@ -34,7 +34,8 @@
                         Placeholder="ex. https://bitwarden.company.com"
                         StyleClass="box-value"
                         ReturnType="Go"
-                        ReturnCommand="{Binding SubmitCommand}" />
+                        ReturnCommand="{Binding SubmitCommand}"
+                        AutomationId="ServerUrlEntry"/>
```
Comment (mpbw2) [inline src/App/Pages/Accounts/EnvironmentPage.xaml:38]: For formatting consistency, there should be a space before `/>`

File: src/App/Pages/Accounts/HomePage.xaml [hunk 4]
```diff
@@ -49,7 +49,9 @@
                         Keyboard="Email"
                         StyleClass="box-value"
                         ReturnType="Go"
-                        ReturnCommand="{Binding ContinueCommand}">
+                        ReturnCommand="{Binding ContinueCommand}"
+                        AutomationId="EmailAddressEntry"
+                        >
```
Comment (mpbw2) [inline src/App/Pages/Accounts/HomePage.xaml:54]: All `>` and `/>` should be on the last line of the block (not a new line)

**Pull request merged**